dev-support/ci/xml_to_md.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/python
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Python file to convert XML properties into Markdown
+import os
+import re
+import zipfile
+import xml.etree.ElementTree as ET
+from collections import namedtuple
+from pathlib import Path
+import sys
+
+Property = namedtuple('Property', ['name', 'value', 'tag', 'description'])
+
+def extract_xml_from_jar(jar_path, xml_filename):
+  xml_files = []
+  with zipfile.ZipFile(jar_path, 'r') as jar:
+    for file_info in jar.infolist():
+      if file_info.filename.endswith(xml_filename):
+        with jar.open(file_info.filename) as xml_file:
+          xml_files.append(xml_file.read())
+  return xml_files
+
+def wrap_config_keys_in_description(description, properties):
+  words = description.split()
+  wrapped_words = []
+  for word in words:
+    clean_word = word.strip('.,()')
+    if clean_word in properties:
+      word = f'`{word}`'
+    wrapped_words.append(word)
+  return ' '.join(wrapped_words)
+
+def parse_xml_file(xml_content, properties):
+  root = ET.fromstring(xml_content)
+  for prop in root.findall('property'):
+    name = prop.findtext('name')
+    if not name:
+      raise ValueError("Property 'name' is required but missing in XML.")
+    description = prop.findtext('description', '')
+    if not description:
+      raise ValueError(f"Property '{name}' is missing a description.")
+    tag = prop.findtext('tag')
+    properties[name] = Property(
+      name = name,
+      value = prop.findtext('value', ''),
+      tag = '<br/>'.join(f'`{t}`' for t in tag.split(', ')),
+      description = wrap_config_keys_in_description(
+        ' '.join(description.split()).strip(),
+        properties
+      )
+    )
+
+
+def generate_markdown(properties):
+  markdown = f"""---
+
+sidebar_label: Appendix
+---
+
+Configuration Key Appendix
+==========================
+
+This page provides a comprehensive overview of all configuration keys available in Ozone.
```

Comment:
Can we add a note that this is auto-generated and should not be edited manually?